app/addons/documents/resources.js

```diff
@@ -85,18 +86,15 @@ Documents.DdocInfo = FauxtonAPI.Model.extend({
 
 Documents.NewDoc = Documents.Doc.extend({
   fetch: function () {
-    var uuid = new Documents.UUID();
-    var deferred = this.deferred = $.Deferred();
-    var that = this;
-
-    uuid.fetch().done(function () {
-      that.set("_id", uuid.next());
-      deferred.resolve();
+    const uuid = new Documents.UUID();
+    const promise = new FauxtonAPI.Promise((resolve) => {
+      uuid.fetch().done(() => {
```

Comment:
Could we use `then` instead of done?

Reply:
Ideally, we need the UUID generation to be extracted out of the model and just be a nice function.